src/EventStore.Core/Messaging/ContinuationEnvelope.cs

```diff
@@ -36,4 +36,22 @@ public class ContinuationEnvelope : IEnvelope {
 			catch (OperationCanceledException) {}
 		}
 	}
+
+	public class AsyncTaskEnvelope : IEnvelope {
+		private readonly Func<Message, CancellationToken, Task> _onMessage;
+		private readonly CancellationToken _cancellationToken;
+
+		public AsyncTaskEnvelope(Func<Message, CancellationToken, Task> onMessage, CancellationToken cancellationToken) {
+			_onMessage = onMessage;
+			_cancellationToken = cancellationToken;
+		}
+
+		public void ReplyWith<T>(T message) where T : Message {
+			try {
+				_onMessage(message, _cancellationToken);
+			}
+			catch (ObjectDisposedException) {}
```

Comment:
probably this catch was for the semaphore, we might not need it here?

Reply:
fixed :+1: